iFixit/react-commerce
# Add product list template

closes #572 

To reduce code duplication for routes I'm proposing to introduce a `templates` folder. We can add a "template" for each kind of page that we have (e.g. product list, product, community, etc).
The advantages of this approach are:

- **Reduced code duplication**: the logic to handle similar routes is handled in a single place
- **Code colocation**: We can locate components that are tightly coupled with the template under that template folder (doing these with `pages` folder was not possible because these files would also become routes). Note that in an effort to keep this PR short I'm not moving `/frontend/components/product-list` components under `/frontend/templates/product-list`. I'll do this in a separate PR

## QA

1. Open Vercel preview
2. Verify that `/Parts` renders correctly
3. Verify that device part and device item type parts render correctly 
4. Verify that devices or item types that don't exist return a 404 page
5. Verify that devices handles typed in the wrong case are redirected to the canonical page
6. Verify that `/Tools` renders correctly
7. Verify that tool categories render correctly
8. Verify that tool categories that don't exist return a 404 page
9. Verify that tool categories handles typed in the wrong case are redirected to the canonical page

cc @danielbeardsley @sterlinghirsh @masonmcelvain
